jvm/src/common/Language/Java/Unsafe.hs

```diff
@@ -273,30 +277,51 @@ instance Coercible (Choice.Choice a) where
   coerce = coerce . Choice.toBool
   unsafeUncoerce = Choice.fromBool . unsafeUncoerce
 
+-- | Inject a value (of primitive or reference type) to a 'JValue'. This
+-- datatype is useful for e.g. passing arguments as a list of homogeneous type.
+-- Synonym for 'coerce'.
+jvalue :: (ty ~ Ty a, Coercible a) => a -> JValue
```

Comment:
Why have the `ty` type variable here?

Reply:
This is preexisting code that was moved earlier in the file because of the intervening splice leading to scoping errors. So I left it as-is. You're right that `ty` here serves no purpose. Probably a holdover from the past.